app/src/main/java/org/literacyapp/battery/BatteryActivity.java

```diff
@@ -2,37 +2,18 @@
 
 import android.app.Activity;
 import android.content.Intent;
-import android.content.IntentFilter;
-import android.os.BatteryManager;
 import android.os.Bundle;
-import android.util.Log;
-import android.widget.TextView;
 
 
 public class BatteryActivity extends Activity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
-        Log.i(getClass().getName(), "onCreate");
         super.onCreate(savedInstanceState);
-
         setContentView(R.layout.activity_battery);
-
-        TextView mcheckBatteryStatus = (TextView) findViewById(R.id.checkBatteryStatus);
-        mcheckBatteryStatus.setText(String.valueOf(checkBatteryStatus()));
-
-        Log.d("BATTERY: ", String.valueOf(checkBatteryStatus()));
+        Intent intent = new Intent();
+        intent.setAction("literacyapp.intent.action.BATTERY_STATUS_CHANGED");
+        sendBroadcast(intent);
```

Comment:
It depends on how often battery level changes are reported via Intents from the Android system. If it happens rarely, the best solution might be to create a background job running periodic checks.